### PR TITLE
MAP-1942 redis 8 upgrade and esLint upgrade as per TEMPLATE TYPESCRIPT

### PR DESCRIPTION
**Pull request by marcusaleman**
As a warden of hmpps-locations-inside-prison we need to update Typescript code to align to the last dependencies update and modernise code with Redis 8.

Reference Template Typescript https://github.com/ministryofjustice/hmpps-template-typescript/pull/492/

Code has been tested locally.

### CODE REVIEW

File: server/services/manageUsersService.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default class ManageUsersService {
     if (username) {
       try {
         return await this.manageUsersApiClient.users.get(token, { username })
-      } catch (e) {
+      } catch {
```
Comment (marcusaleman) [inline server/services/manageUsersService.ts:10]: As upgraded Lint claim about unused 'e' variable

File: server/middleware/setUpWebSession.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { v4 as uuidv4 } from 'uuid'
 import session, { MemoryStore, Store } from 'express-session'
-import RedisStore from 'connect-redis'
+import { RedisStore } from 'connect-redis'
```
Comment (marcusaleman) [inline server/middleware/setUpWebSession.ts:3]: As per https://github.com/ministryofjustice/hmpps-template-typescript/pull/492

File: server/middleware/setUpHealthChecks.ts [hunk 3]
```diff
@@ -15,11 +15,11 @@ export default function setUpHealthChecks(applicationInfo: ApplicationInfo): Rou
...
   })
 
-  router.get('/ping', (req, res) =>
+  router.get('/ping', (req, res) => {
     res.send({
       status: 'UP',
-    }),
-  )
+    })
+  })
```
Comment (marcusaleman) [inline server/middleware/setUpHealthChecks.ts:23]: As per https://github.com/ministryofjustice/hmpps-template-typescript/pull/448

File: server/services/manageUsersService.ts (hunk 1 above)
Comment (jnioche-jd) [reply server/services/manageUsersService.ts:10]: what about logging the exception instead?

File: package.json [hunk 4]
```diff
@@ -97,7 +97,7 @@
     "bunyan-format": "^0.2.1",
     "compression": "^1.7.5",
     "connect-flash": "^0.1.1",
-    "connect-redis": "^7.1.1",
+    "connect-redis": "^8.0.1",
```
Comment (jnioche-jd) [inline package.json:100]: the docker compose file specifies a 7.* version, needs fixing too

**Review by jnioche-jd — CHANGES_REQUESTED** [review]
see comment

**Comment by marcusaleman** [comment]
**CHANGE THIS AS A DRAFT ** Added a logger(e) for some scenario however I need to verify that the error doesn't print Creds

File: package.json (hunk 4 above)
Comment (jnioche-jd) [reply package.json:100]: ok so prod uses 7.2 but the template requies connect-redis 8 which is a bit of a discrepancy. let's leave docker-compose as it is

**Review by Thource — APPROVED** [review]

**Pull request merged**